projects/plugins/jetpack/extensions/blocks/subscriptions/panel.js

```diff
@@ -80,23 +89,40 @@ export default function SubscribePanels() {
 			>
 				{ showNotices && (
 					<InspectorNotice>
-						{ createInterpolateElement(
-							sprintf(
-								/* translators: 1$s will be subscribers, %2$s will be social followers */
-								__( 'This post will reach <span>%1$s</span> and <span>%2$s</span>.', 'jetpack' ),
+						{ willShareToSocialMedia &&
+							createInterpolateElement(
 								sprintf(
-									/* translators: %s will be a number of subscribers */
-									_n( '%s subscriber', '%s subscribers', subscriberCount, 'jetpack' ),
-									numberFormat( subscriberCount )
+									/* translators: 1$s will be subscribers, %2$s will be social followers */
+									__( 'This post will reach <span>%1$s</span> and <span>%2$s</span>.', 'jetpack' ),
+									sprintf(
+										/* translators: %s will be a number of subscribers */
+										_n( '%s subscriber', '%s subscribers', subscriberCount, 'jetpack' ),
+										numberFormat( subscriberCount )
+									),
+									sprintf(
+										/* translators: %s will be a number of social followers */
+										_n( '%s social follower', '%s social followers', followerCount, 'jetpack' ),
+										numberFormat( followerCount )
+									)
 								),
+								{ span: <span className="jetpack-subscribe-reader-count" /> }
+							) }
+						{ ! willShareToSocialMedia &&
+							createInterpolateElement(
 								sprintf(
-									/* translators: %s will be a number of social followers */
-									_n( '%s social follower', '%s social followers', followerCount, 'jetpack' ),
-									numberFormat( followerCount )
-								)
-							),
-							{ span: <span className="jetpack-subscribe-reader-count" /> }
-						) }
+									/* translators: 1$s will be subscribers, %2$s will be social followers */
+									__(
+										'This post will reach <span>%1$s</span> and wonʼt be shared on social.',
```

Comment:
This feels more understandable to me, but maybe that's just me :)

```suggestion
										'This post will reach <span>%1$s</span> and wonʼt be shared on social media.',
```